ocs_ci/ocs/resources/pod.py

```diff
@@ -143,6 +145,58 @@ def exec_ceph_cmd(self, ceph_cmd, format='json-pretty'):
             return [item for item in out if item]
         return out
 
+    def run_io(
+        self, storage_type, size, direction='rw', workers=1, runtime=60,
+        run_in_bg=True
+    ):
+        """
+        Execute FIO on a pod
+
+        Args:
+             storage_type (str): 'fs' or 'block'
+             size (str): Size in MB, e.g. '200M'
+             direction (str): Determines the operation:
+                'ro', 'wo', 'rw' (default: 'rw')
+            workers (int): Number of threads to execute FIO
+            runtime (int): Number of seconds IO should run for
+            run_in_bg (bool): True in case the FIO should run in background,
+                False otherwise
+        """
+        def execute_fio():
+            name = 'test_workload'
+            spec = self.pod_data.get('spec')
+            path = (
+                spec.get('containers')[0].get('volumeMounts')[0].get(
+                    'mountPath'
+                )
+            )
+            work_load = 'fio'
+            # few io parameters for Fio
+
+            wl = workload.WorkLoad(
+                name, path, work_load, storage_type, self, workers
+            )
+            assert wl.setup()
+            if direction == 'rw':
+                io_params = templating.load_yaml_to_dict(
+                    constants.FIO_IO_RW_PARAMS_YAML
+                )
+            else:
+                io_params = templating.load_yaml_to_dict(
+                    constants.FIO_IO_PARAMS_YAML
+                )
+            io_params['runtime'] = runtime
+            io_params['size'] = size
+
+            future_result = wl.run(**io_params)
+            logger.info(future_result.result())
+
+        if run_in_bg:
```

Comment:
I am not sure how this can work in fg, the workload essentially kicks off in bg.

Reply:
right,
The intention is to not block the main thread the is executing the tests (so we will be able to disrupt if we want for example).
Do you think this var name is confusing? any suggestion?

Reply:
I see now, the execute_fio is serial due to .result(),  we can leave this as is. thanks